c/frame/__repr__.cc

```diff
@@ -75,135 +82,135 @@ static bool in_jupyter() {
-        html << "</th>";
       }
       html << "</tr>\n";
     }
 
-    void render_column_types() {
+    void _render_column_types() {
       html << "    <tr class='coltypes'>";
-      html << "<td class='row_index'></td>";
-      for (size_t j = 0; j < ncols; ++j) {
-        if (j == cols0) {
-          j = ncols - cols1;
+      if (render_row_indices_) {
+        html << "<td class='row_index'></td>";
+      }
+      for (size_t j : colindices_) {
+        if (j == dt::NA_index) {
           html << "<td></td>";
+        } else {
+          auto stype_info = info(dt_->get_column(j).stype());
+          size_t elemsize = stype_info.elemsize();
+          html << "<td class='" << stype_info.ltype_name()
+               << "' title='" << stype_info.name() << "'>";
+          for (size_t k = 0; k < elemsize; ++k) html << "&#x25AA;";
+          html << "</td>";
         }
-        SType stype = dt->get_column(j).stype();
-        size_t elemsize = info(stype).elemsize();
-        html << "<td class='" << info(stype).ltype_name()
-             << "' title='" << info(stype).name() << "'>";
-        for (size_t k = 0; k < elemsize; ++k) html << "&#x25AA;";
-        html << "</td>";
       }
       html << "</tr>\n";
     }
 
-    void render_table_body() {
-      html << "  <tbody>\n";
-      for (size_t i = 0; i < nrows; ++i) {
-        if (i == rows0) {
-          i = nrows - rows1;
-          render_ellipsis_row();
+    void _render_data_rows() {
+      for (size_t i : rowindices_) {
+        if (i == dt::NA_index) {
+          _render_ellipsis_row();
+        } else {
+          _render_data_row(i);
         }
-        render_data_row(i);
       }
-      html << "  </tbody>\n";
     }
 
-    void render_ellipsis_row() {
+    void _render_ellipsis_row() {
       html << "    <tr>";
-      html << "<td class='row_index'>&#x22EE;</td>";
-      for (size_t j = 0; j < ncols; ++j) {
-        if (j == cols0) {
-          j = ncols - cols1;
+      if (render_row_indices_) {
+        html << "<td class='row_index'>&#x22EE;</td>";
+      }
+      for (size_t j : colindices_) {
+        if (j == dt::NA_index) {
           html << "<td class='hellipsis'>&#x22F1;</td>";
+        } else {
+          html << "<td class='hellipsis'>&#x22EE;</td>";
         }
-        html << "<td class='hellipsis'>&#x22EE;</td>";
       }
       html << "</tr>\n";
     }
 
-    void render_data_row(size_t i) {
+    void _render_data_row(size_t i) {
       html << "    <tr>";
-      html << "<td class='row_index'>";
-      render_comma_separated(i);
-      html << "</td>";
-      for (size_t j = 0; j < ncols; ++j) {
-        if (j == cols0) {
-          j = ncols - cols1;
+      if (render_row_indices_) {
+        html << "<td class='row_index'>";
+        _render_comma_separated(i);
+        html << "</td>";
+      }
+      for (size_t j : colindices_) {
+        if (j == dt::NA_index) {
           html << "<td class=vellipsis>&hellip;</td>";
```

Comment:
Adding single-quotes will make it consistent with the rest of the code: `class='vellipsis'`

Reply:
sure, I can do it with the next PR